service/storage_service.cc

```diff
@@ -2013,11 +2013,19 @@ future<> storage_service::decommission() {
             bool raft_available = ss._group0->wait_for_raft().get();
             bool left_token_ring = false;
 
-            // Step 2: Prepare to sync data
             std::unordered_set<gms::inet_address> nodes_unknown_verb;
             std::unordered_set<gms::inet_address> nodes_down;
             auto req = node_ops_cmd_request{node_ops_cmd::decommission_prepare, uuid, ignore_nodes, leaving_nodes, {}};
             try {
+                // Step 2: Start heartbeat updater
+                auto heartbeat_updater_done = make_lw_shared<bool>(false);
```

Comment:
orthogonal to this fix, it looks like `heartbeat_updater_done` should be a simple `abort_source`